core/meta/src/TClass.cxx

```diff
@@ -3455,38 +3455,32 @@ const char *TClass::GetSharedLibs()
 
 TList *TClass::GetListOfBases()
 {
-   if (!fBase) {
-      if (fCanLoadClassInfo) {
-         if (fState == kHasTClassInit) {
+   if (fBase)
+      return fBase;
 
-            R__LOCKGUARD(gInterpreterMutex);
-            // NOTE: Add test to prevent redo if another thread has already done the work.
-            // if (!fHasRootPcmInfo) {
+   if (fState == kHasTClassInit) {
```

Comment:
There is a noticeable change in behavior that is not explained.  I.e. now GetProtoNorm and FillTClass is called whether or not fCanLoadClassInfo is true or false.  There need to be a clarification on why this is fine and/or as performant.

Reply:
Note that after a recent change, we now set `fCanLoadClassInfo` to `false` once the information is already loaded, so the `if (fCanLoadClassInfo)` around the `GetProtoNorm()` will probably have the opposite effect than intended. The right check really is only if the class has the `Init()` function, which will be used by the proto class to fill a `TClass` with the information if needed. That said, `FillTClass()` is a quite large function, so we want to avoid calling it if we can...

Reply:
> will probably have the opposite effect than intended.

I am not understanding this statement.  A way to semantically write the existing code is
```
 if (fCanLoadClassInfo)
    LoadClassInfo() unless we have a TProtoClass in which case we take the info from there.
```
if fCanLoadClassInfo is false either the information is already loaded/present-in-memory or the information does not exist (can not be loaded).

Completely related in line 3470 the call to LoadClassInfo is now no longer protected against the case where the information in present in the TProtoClass.  This is a **significant memory performance degradation** and essentially revert bf6f2f34e7950f7c888e6112f97ec64cefe7cd08 as it now lead to the header being **always** loaded in memory and parsed when the list of base class is requested.